src/declarations/stencil-public-compiler.ts

```diff
@@ -1255,10 +1259,30 @@ export interface ResolveModuleIdResults {
   pkgDirPath: string;
 }
 
+/**
+ * A controller which provides for communication and coordination between
+ * threaded workers.
+ */
 export interface WorkerMainController {
+  /**
+   * Send a given set of arguments to a worker
+   */
   send(...args: any[]): Promise<any>;
-  handler(name: string): (...args: any[]) => Promise<any>;
+  /**
+   * Handle a particular method
+   *
+   * @param name of the method to be passed to a worker
+   * @returns a Promise wrapping the results
+   */
+  handler(name: WorkerContextMethod): (...args: any[]) => Promise<any>;
```

Comment:
I'm not sure we can narrow this type without it technically constituting a breaking change (with respect to typings) - we expose this through our Node API on `CompilerSystem`.

Reply:
mmmm right...I'll put that in another PR that we can merge in the future or something

I'd be surprised if this caused too many issues for people though, so if we're open to shipping a small breaking change in a 4.8 release we could leave it in here

Reply:
> I'd be surprised if this caused too many issues for people though, so if we're open to shipping a small breaking change in a 4.8 release we could leave it in here

Yeah, I doubt it'd be _too_ widespread. However, out of abundance of caution, can we slate that change for v5 and add a TODO so we don't forget?

Reply:
Sure, I'll put in a TODO

Reply:
Just added `STENCIL-1005`